core/trino-main/src/main/java/io/trino/execution/SetTimeZoneTask.java

```diff
@@ -78,7 +78,7 @@ public ListenableFuture<Void> execute(
     {
         String timeZoneId = statement.getTimeZone()
                 .map(timeZone -> getTimeZoneId(timeZone, statement, stateMachine, parameters, warningCollector))
-                .orElse(TimeZone.getDefault().getID());
+                .orElseGet(() -> TimeZone.getDefault().getID());
```

Comment:
The new code is faster

```
Benchmark                                     Mode  Cnt  Score   Error  Units
BenchmarkOptional.orElseDefaultTimeZoneId     avgt  200  7.078 ± 0.024  ns/op
BenchmarkOptional.orElseGetDefaultTimeZoneId  avgt  200  4.053 ± 0.020  ns/op
```

however, i am reverting this.
I think the code is incorrect and don't want to touch it now

Reply:
reported https://github.com/trinodb/trino/issues/15314